chapter6/exercise3.js

```diff
@@ -0,0 +1,77 @@
+(function() {
```

Comment:
generalnie chyba nie do konca zrozumiales koncept tego zadania. chodzilo o to, zeby stworzyc obiekty typow ArraySeq i RangeSeq ktore beda posiadac wspolny interfejs do iteracji wykorzystywany przez logFive.

+ logFive ma nie wiedziec nic o obiekcie, ktory jest do niego przekazywany poza tym, ze zawiera metody umozliwiajace na iteracje
+ oba obiekty musza implementowac interfejs iteracyjny i posiadac metody takie jak: hasNext(), next(), current()


oprocz metody "next" dodaj rowniez metode "hasNext". wykorzystaj ja w metodzie "next".

staraj sie zawsze rozbijac elementy wg zasady single responsibility - niech jedna metoda posiada w sobie logike tylko do jednego zadania. w tym wypadku moze to byc nieco na wyrost, ale bedziesz miala dwie metody i dwa zadania:
. hasNext - sprawdzenie czy kolejny element istnieje
. next - uzyskanie kolejnego elementu

dzieki temu mozesz sprawdzic czy masz kolejny element bez przymusu przesuwania "wskaznika".

jest to dosc znany pattern zwany iterator, ktory uzywany jest np w javie -
https://www.tutorialspoint.com/java/java_using_iterator.htm